src/OpenTelemetry.Contrib.Instrumentation.Runtime/Implementation/GcInstrumentation.cs

```diff
@@ -0,0 +1,102 @@
+        private const string GcTimeCounterName = "time-in-gc";
+        private const string Gen0SizeCounterName = "gen-0-size";
+        private const string Gen1SizeCounterName = "gen-1-size";
+        private const string Gen2SizeCounterName = "gen-2-size";
+        private const string LohSizeCounterName = "loh-size";
+        private const string PohSizeCounterName = "poh-size";
+
+        private readonly Meter meter;
+        private readonly IEventCounterStore eventCounterStore;
+        private readonly ObservableCounter<double> gcHeapSizeCounter;
+        private readonly ObservableGauge<int> gen0GCCounter;
+        private readonly ObservableGauge<int> gen1GCCounter;
+        private readonly ObservableGauge<int> gen2GCCounter;
+        private readonly ObservableGauge<long> allocRateCounter;
+        private readonly ObservableCounter<double> fragmentationCounter;
+        private readonly ObservableCounter<double> committedCounter;
+        private readonly ObservableCounter<double> gcTimeCounter;
+        private readonly ObservableCounter<long> gen0SizeCounter;
+        private readonly ObservableCounter<long> gen1SizeCounter;
+        private readonly ObservableCounter<long> gen2SizeCounter;
+        private readonly ObservableCounter<long> lohSizeCounter;
+        private readonly ObservableCounter<long> pohSizeCounter;
+
+        public GcInstrumentation(Meter meter, IEventCounterStore eventCounterStore)
+        {
+            this.meter = meter;
+            this.eventCounterStore = eventCounterStore;
+
+            this.eventCounterStore.Subscribe(GcTimeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen0SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen1SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(Gen2SizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(LohSizeCounterName, EventCounterType.Mean);
+            this.eventCounterStore.Subscribe(PohSizeCounterName, EventCounterType.Mean);
+
+            this.gcHeapSizeCounter = meter.CreateObservableCounter($"{RuntimeMetrics.MetricPrefix}gc_heap_size", () => (double)(GC.GetTotalMemory(false) / 1_000_000), "MB", "GC Heap Size");
+            this.gen0GCCounter = meter.CreateObservableGauge($"{RuntimeMetrics.MetricPrefix}gen_0-gc_count", () => GC.CollectionCount(0), description: "Gen 0 GC Count");
+            this.gen1GCCounter = meter.CreateObservableGauge($"{RuntimeMetrics.MetricPrefix}gen_1-gc_count", () => GC.CollectionCount(1), description: "Gen 1 GC Count");
+            this.gen2GCCounter = meter.CreateObservableGauge($"{RuntimeMetrics.MetricPrefix}gen_2-gc_count", () => GC.CollectionCount(2), description: "Gen 2 GC Count");
+            this.allocRateCounter = meter.CreateObservableGauge($"{RuntimeMetrics.MetricPrefix}alloc_rate", () => GC.GetTotalAllocatedBytes(), "B", "Allocation Rate");
```

Comment:
Questions again here, `GC.GetTotalAllocatedBytes` is `IncrementingPollingCounter`
[RuntimeEventSource ](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs)

From the docs - Provides a variant of `EventCounter` for variables that are ever-increasing, such as the number of exceptions in the runtime

Why not to use `ObservableCounter`?

Reply:
Fixed.